crates/viewer/re_redap_browser/src/dataset_ui.rs

```diff
@@ -94,35 +110,40 @@ pub fn dataset_ui(
             right: 16,
         })
         .show(ui, |ui| {
-            ui.horizontal(|ui| {
-                ui.heading(RichText::new(dataset.name()).strong());
-                if ui.small_icon_button(&icons::RESET).clicked() {
-                    let _ = ctx
-                        .command_sender
-                        .send(Command::RefreshCollection(origin.clone()));
-                }
-            });
+            egui::Sides::new().show(
+                ui,
+                |ui| {
+                    ui.heading(RichText::new(dataset.name()).strong());
+                    if ui.small_icon_button(&icons::RESET).clicked() {
+                        let _ = ctx
+                            .command_sender
+                            .send(Command::RefreshCollection(origin.clone()));
+                    }
+                },
+                |ui| {
+                    table_delegate.table_config.button_ui(ui);
+                },
+            );
         });
 
     apply_table_style_fixes(ui.style_mut());
 
     egui_table::Table::new()
         .id_salt(table_id_salt)
         .columns(
-            columns
-                .iter()
-                .map(|field| {
-                    egui_table::Column::new(200.0)
-                        .resizable(true)
-                        .id(egui::Id::new(field))
-                })
+            table_delegate
+                .table_config
+                .visible_column_ids()
+                .map(|id| egui_table::Column::new(200.0).resizable(true).id(id))
```

Comment:
This is the kind of delegate <-> table config thing that could be more tightly integrated.